tests/unit/string_util_test.cpp

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/common/string_util.h>
+}
+
+TEST(strstr, basic)
+{
+   const char haystack[] = "haystack", needle[] = "needle", \
+   hay[] = "hay", empty[] = "";
+   ASSERT_EQ(strstr(empty, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, needle), (char *) NULL);
+   ASSERT_EQ(strstr(haystack, hay), haystack);
+}
+
+TEST(strncpy, basic)
+{
+   char dest[4], src[] = "a";
+   strncpy(dest, src,2);
+   ASSERT_EQ(0, strcmp(dest, "a"));
+}
+
+TEST(strncat, basic)
+{
+   char dest[6] = "abc", src[] = "d";
+   strncat(dest, src, 4);
+   ASSERT_EQ(0, strcmp(dest, "abcd"));
```

Comment:
I believe there is something like ASSERT_STREQ, which is way more convenient than this.

Reply:
You're right. Fixed this.